.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,28 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
```

Comment:
I'm not sure if opening a PR should trigger a new image build - it seems to me, that it would be better to build new version only after PR is merged